BackEndLib/Wchar.cpp

```diff
@@ -885,3 +893,45 @@ WSTRING WCSReplace(
 		processed = loc + from.size();
 	}
 }
+
+//*****************************************************************************
+WSTRING WCSToLower(WSTRING const &source)
+//Converts WSTRING to lowercase
+{
+	WSTRING lowercased;
+	lowercased.reserve(source.size());
+	for (WSTRING::const_iterator it = source.begin(); it != source.end(); ++it)
+		lowercased += towlower(*it);
+
+	return lowercased;
+}
+
+//*****************************************************************************
+const std::vector<WSTRING> WCSExplode(WSTRING const &source, WCHAR const delimiter)
+// Explodes a string into pieces
+// Adapted from: https://stackoverflow.com/a/12967010
+{
+	std::vector<WSTRING> result;
+	std::wistringstream iss(source);
+
+	for (WSTRING token; std::getline(iss, token, delimiter); )
+	{
+		if (!token.empty())
+			result.push_back(token);
+	}
+
+	return result;
+}
+
+//*****************************************************************************
+bool WCSContainsAll(WSTRING const &haystack, std::vector<WSTRING> const &needles)
+// Returns true if haystack contains every string in needle, even if they overlap
+{
+	for (std::vector<WSTRING>::const_iterator iter = needles.begin(); iter < needles.end(); ++iter)
+	{
+		if (haystack.find(*iter) == WSTRING::npos)
+			return false;
+	}
+
+	return true;
```

Comment:
From `<algorithm>`, C++11:

```cpp
	return std::all_of(needles.begin(), needles.end(),
		[&haystack]
		{
			return haystack.find(*iter) != WSTRING::npos;
		});
```

Consider `std::none_of` as well, might be more readable due to the avoided negation in the body of the lambda expression.